trunk/src/lsc/myloopdef.py

```diff
@@ -897,6 +903,49 @@ def checkpsf(imglist, database='photlco'):
         else:
             print 'status ' + str(status) + ': unknown status'
 
+
+def make_psf_plot(psf_filename):
+    """
+    Displays plots of PSFs for the checkpsf stage without using iraf
+    :param psf_filename: filepath+filename of psf file
+    """
+    plt.clf()
+
+    psf_hdul = fits.open(psf_filename)
+    N = psf_hdul[0].header['PSFHEIGH'] / psf_hdul[0].header['NPSFSTAR']
+    sigma_x = psf_hdul[0].header['PAR1']
+    sigma_y = psf_hdul[0].header['PAR2']
+    psfrad = psf_hdul[0].header['PSFRAD']
+    NAXIS1 = psf_hdul[0].header['NAXIS1']
+    NAXIS2 = psf_hdul[0].header['NAXIS2']
+
+    x = np.linspace(-psfrad, psfrad, num=NAXIS1)
+    y = np.linspace(-psfrad, psfrad, num=NAXIS2)
+    X, Y = np.meshgrid(x, y)
+    # PSF is elliptical gaussian (from header) + residuals (from img data)
+    # in description https://iraf.net/irafhelp.php?val=seepsf
+    # not 100% sure normalization is correct, but tested on
+    #       good and bad psfs and I think it's right
+    analytic = N * np.exp(-(((X ** 2) / (sigma_x ** 2)) + ((Y ** 2) / (sigma_y ** 2))) / 2)
+    residual = psf_hdul[0].data
+    Z = analytic + residual
+
+    ax = plt.subplot(1, 1, 1, projection='3d')
+    """
+    # the transparency makes this challenging to interpret
+    ax.plot_wireframe(X, Y, Z, rcount=2 * psfrad + 1, ccount=2 * psfrad + 1)
+    """
+    # replicate iraf look, much slower than wireframe
+    ax.plot_surface(X,Y,Z,rcount=2*psfrad+1,ccount=2*psfrad+1,
+            antialiased=True,linewidth=.25,color='black',edgecolor='white')
+    
+
+    ax.view_init(elev=40, azim=330)  # replicating starting view of iraf PSF
+    ax.set_axis_off()
+    ax.set_title('PSF for {psf_filename}'.format(psf_filename=psf_filename))
+    plt.gcf().set_size_inches(8,4)
+    plt.gcf().tight_layout()
```

Comment:
Instead of calling `plt.gcf()`, why don't you use `plt.subplots` (plural) above, which returns the figure object and the axes. You can also set the figure size directly in that call. (Or maybe this doesn't work for 3D axes? Not sure.)

Reply:
I think plt.gcf() will continue to plot in an open window, where plt.subplots will make a new figure window every time, which is a pain when you're processing a bunch of data

Reply:
Ahh great point. Never mind then.